### ISSUE TITLE
`patch.Helper` doesn't work with secrets and configmaps

### ISSUE THREAD

**Issue by errordeveloper** [fluxcd/pkg#588]
I was using patcher in a project and discovered the hard way that it doesn't actually patch secrets.

The reason behind that is to do with this logic:
https://github.com/fluxcd/pkg/blob/a04ccbfc5bc8348e31131978140640f0b1363d18/runtime/patch/patch.go#L240

So it very deliberately checks `spec` field, which is something secrets and configmaps don't have.

**Comment by stefanprodan** [comment]
I suggest you use ssa.apply to patch Kubernetes native resources, the patcher is for custom resources which subscribe to the spec/status standard.

**Comment by errordeveloper** [comment]
@stefanprodan maybe that should be made more explicit in the docs?

**Comment by stefanprodan** [comment]
Yes for sure, patcher is for controllers to set finalizers and status conditions, we use it as such in Flux, and we should clearly state it’s purpose in docs.
